lib/workflows/vm-migration/begin.js

```diff
@@ -214,9 +217,24 @@ function createProvisionPayload(job, cb) {
     delete vmPayload.pid;
     delete vmPayload.tmpfs;
 
+    // Handle delegated datasets for SmartOS and LX zones.
+    if (Array.isArray(vmPayload.datasets) && vmPayload.datasets.length > 0) {
+        assert.ok(brand === 'joyent' || brand === 'joyent-minimal' ||
```

Comment:
Ditto, I wouldn't assert here. Check the brand and return an error if it's not the expected value.

Reply:
I will change to a callback error.